server/app/services/program/ProgramService.java

```diff
@@ -258,7 +266,7 @@ public CompletionStage<ProgramDefinition> getActiveFullProgramDefinitionAsync(
   public ProgramDefinition getDraftFullProgramDefinition(String programSlug)
       throws ProgramDraftNotFoundException {
     ProgramModel draftProgram = programRepository.getDraftProgramFromSlug(programSlug);
-    return syncProgramAssociations(draftProgram).toCompletableFuture().join();
+    return syncProgramAssociationsAndSaveToCache(draftProgram).toCompletableFuture().join();
```

Comment:
**suggestion:** Maybe the function should be called `syncProgramAssociationsAndMaybeSaveToCache`, since we only save active programs, but could also be explained in javadoc I guess. Its nice the name exposes that though - but it does it in a function layered way down, so may want to update the name of the other one that actually saves it

Reply:
I don't like "maybe" cause then you have to read the function anyway, so how about `syncProgramAssociationsAndSaveToCacheIfActive()`?